From d2480552addba8c95869afbf75dd9ab492fa6aa5 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Sat, 19 Aug 2017 02:30:15 +0800
Subject: [PATCH 2451/5242] MLK-16013-42 ARM64: imx8mq-evk: enable typec port
 to be otg

commit  6489e4373bff76b91b703eaa929157285cdb4390 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add extcon for typec port(USB0) for support dual roles.

Acked-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
index 1dc7e51..5479e06 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk.dts
@@ -422,7 +422,7 @@
 		};
 	};
 
-	ptn5110@50 {
+	typec_ptn5100: ptn5110@50 {
 		compatible = "usb,tcpci";
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_typec>;
@@ -567,7 +567,8 @@
 
 &usb_dwc3_0 {
 	status = "okay";
-	dr_mode = "peripheral";
+	extcon = <&typec_ptn5100>;
+	dr_mode = "otg";
 };
 
 &usb3_phy1 {
-- 
1.7.9.5

